app/models/playlist_item.rb

```diff
@@ -28,17 +28,19 @@ class PlaylistItem < ActiveRecord::Base
   end
 
   def duplicate!
-    new_clip = clip.dup
-    new_clip.save!
+    unless (clip.master_file.nil?)
```

Comment:
We'll definitely need tests, I'll make sure to add them later
I'll make that change. It looks better.